packages/agent/src/did-api.ts

```diff
@@ -0,0 +1,321 @@
+
+import { InMemoryDidStore } from './store-did.js';
+import { DidResolverCacheMemory } from './prototyping/dids/resolver-cache-memory.js';
+
+export enum DidInterface {
+  Create  = 'Create',
+  // Deactivate = 'Deactivate',
+  Resolve = 'Resolve',
+  // Update  = 'Update'
+}
+
+export interface DidMessageParams {
+  [DidInterface.Create]: DidCreateParams;
+  // [DidInterface.Deactivate]: DidDeactivateParams;
+  [DidInterface.Resolve]: DidResolveParams;
+  // [DidInterface.Update]: DidUpdateParams;
+}
+
+export interface DidMessageResult {
+  [DidInterface.Create]: DidCreateResult;
+  // [DidInterface.Deactivate]: DidDeactivateResult;
+  [DidInterface.Resolve]: DidResolveResult;
+  // [DidInterface.Update]: DidUpdateResult;
+}
+
+export type DidCreateResult = {
+  uri: string;
+  document: DidDocument;
+  metadata: DidMetadata;
+}
+
+export type DidResolveResult = DidResolutionResult
+
+export type DidRequest<T extends DidInterface> = {
+  messageType: T;
+  messageParams: DidMessageParams[T];
+}
+
+export type DidResolveParams = {
+  didUri: string;
+  options?: DidResolutionOptions;
+}
+
+export type DidResponse<T extends DidInterface> = ResponseStatus & {
+  result?: DidMessageResult[T];
+};
+
+export interface DidCreateParams<
+  TKeyManager = AgentKeyManager,
+  TMethod extends keyof DidMethodCreateOptions<TKeyManager> = keyof DidMethodCreateOptions<TKeyManager>
+> {
+  method: TMethod;
+  options?: DidMethodCreateOptions<TKeyManager>[TMethod];
+  store?: boolean;
+  tenant?: string;
+}
+
+export interface DidMethodCreateOptions<TKeyManager> {
+  dht: DidDhtCreateOptions<TKeyManager>;
+  jwk: DidJwkCreateOptions<TKeyManager>;
+}
+
+export interface DidApiParams {
+  didMethods: DidMethodApi[];
+
+  agent?: Web5PlatformAgent;
+
+  /**
+   * An optional `DidResolverCache` instance used for caching resolved DID documents.
+   *
+   * Providing a cache implementation can significantly enhance resolution performance by avoiding
+   * redundant resolutions for previously resolved DIDs. If omitted, a no-operation cache is used,
+   * which effectively disables caching.
+   */
+  resolverCache?: DidResolverCache;
+
+  store?: AgentDataStore<PortableDid>;
+}
+
+export function isDidRequest<T extends DidInterface>(
+  didRequest: DidRequest<DidInterface>, messageType: T
+): didRequest is DidRequest<T> {
+  return didRequest.messageType === messageType;
+}
+
+export class AgentDidApi<TKeyManager extends AgentKeyManager = AgentKeyManager> extends DidResolver {
+  /**
+   * Holds the instance of a `Web5PlatformAgent` that represents the current execution context for
+   * the `AgentDidApi`. This agent is used to interact with other Web5 agent components. It's vital
+   * to ensure this instance is set to correctly contextualize operations within the broader Web5
+   * Agent framework.
+   */
+  private _agent?: Web5PlatformAgent;
+
+  private _didMethods: Map<string, DidMethodApi> = new Map();
```

Comment:
Trying to figure out how to use this. If two methods are declared here what does it do? Maybe a comment for all of the params would help understand what to do with all of them